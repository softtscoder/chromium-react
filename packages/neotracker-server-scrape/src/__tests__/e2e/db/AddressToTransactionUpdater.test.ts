import {
  AddressToTransactionRevert,
  AddressToTransactionSave,
  AddressToTransactionUpdater,
} from '../../../db/AddressToTransactionUpdater';
import { updaterUnitTest, UpdaterUtilTestOptions } from '../../data/updaterTestUtil';

const getInputs = (): AddressToTransactionSave => ({
  transactions: [
    {
      addressIDs: ['APyEx5f4Zm4oCHwFWiSTaph1fPBxZacYVR'],
      transactionID: '12341234',
    },
  ],
});

const getSecondaryInputs = (): AddressToTransactionSave => ({
  transactions: [
    {
      addressIDs: ['GIoTx5f4Zm4oCHww38hif0vofPBxZacYRT'],
      transactionID: '23452345',
    },
  ],
});

const getReversions = (): AddressToTransactionRevert => ({
  transactionIDs: ['23452345'],
});

const utilOptions: UpdaterUtilTestOptions<AddressToTransactionSave, AddressToTransactionRevert, undefined> = {
  name: `AddressToTransaction Updater`,
  createUpdater: () => new AddressToTransactionUpdater(),
  getInputs,
  getSecondaryInputs,
  getReversions,
};

updaterUnitTest(utilOptions);
