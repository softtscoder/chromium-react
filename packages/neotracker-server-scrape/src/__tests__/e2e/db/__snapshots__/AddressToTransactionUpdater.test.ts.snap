// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressToTransaction Updater AddressToTransaction Updater: handles duplicate inserts/saves 1`] = `
Object {
  "action": Array [],
  "address": Array [],
  "addressToTransaction": Array [
    Object {
      "id1": "APyEx5f4Zm4oCHwFWiSTaph1fPBxZacYVR",
      "id2": "12341234",
    },
  ],
  "addressToTransfer": Array [],
  "asset": Array [],
  "assetToTransaction": Array [],
  "block": Array [],
  "coin": Array [],
  "contract": Array [],
  "dataPoint": Array [],
  "migration": Array [],
  "processedIndex": Array [],
  "transaction": Array [],
  "transactionInputOutput": Array [],
  "transfer": Array [],
}
`;

exports[`AddressToTransaction Updater AddressToTransaction Updater: handles insert/save 1`] = `
Object {
  "action": Array [],
  "address": Array [],
  "addressToTransaction": Array [
    Object {
      "id1": "APyEx5f4Zm4oCHwFWiSTaph1fPBxZacYVR",
      "id2": "12341234",
    },
  ],
  "addressToTransfer": Array [],
  "asset": Array [],
  "assetToTransaction": Array [],
  "block": Array [],
  "coin": Array [],
  "contract": Array [],
  "dataPoint": Array [],
  "migration": Array [],
  "processedIndex": Array [],
  "transaction": Array [],
  "transactionInputOutput": Array [],
  "transfer": Array [],
}
`;

exports[`AddressToTransaction Updater AddressToTransaction Updater: handles reverting inputs/saves 1`] = `
Object {
  "action": Array [],
  "address": Array [],
  "addressToTransaction": Array [
    Object {
      "id1": "APyEx5f4Zm4oCHwFWiSTaph1fPBxZacYVR",
      "id2": "12341234",
    },
  ],
  "addressToTransfer": Array [],
  "asset": Array [],
  "assetToTransaction": Array [],
  "block": Array [],
  "coin": Array [],
  "contract": Array [],
  "dataPoint": Array [],
  "migration": Array [],
  "processedIndex": Array [],
  "transaction": Array [],
  "transactionInputOutput": Array [],
  "transfer": Array [],
}
`;

exports[`AddressToTransaction Updater AddressToTransaction Updater: handles reverting inputs/saves 2`] = `
Object {
  "action": Array [],
  "address": Array [],
  "addressToTransaction": Array [
    Object {
      "id1": "APyEx5f4Zm4oCHwFWiSTaph1fPBxZacYVR",
      "id2": "12341234",
    },
    Object {
      "id1": "GIoTx5f4Zm4oCHww38hif0vofPBxZacYRT",
      "id2": "23452345",
    },
  ],
  "addressToTransfer": Array [],
  "asset": Array [],
  "assetToTransaction": Array [],
  "block": Array [],
  "coin": Array [],
  "contract": Array [],
  "dataPoint": Array [],
  "migration": Array [],
  "processedIndex": Array [],
  "transaction": Array [],
  "transactionInputOutput": Array [],
  "transfer": Array [],
}
`;

exports[`AddressToTransaction Updater AddressToTransaction Updater: handles reverting inputs/saves 3`] = `
Object {
  "action": Array [],
  "address": Array [],
  "addressToTransaction": Array [
    Object {
      "id1": "APyEx5f4Zm4oCHwFWiSTaph1fPBxZacYVR",
      "id2": "12341234",
    },
  ],
  "addressToTransfer": Array [],
  "asset": Array [],
  "assetToTransaction": Array [],
  "block": Array [],
  "coin": Array [],
  "contract": Array [],
  "dataPoint": Array [],
  "migration": Array [],
  "processedIndex": Array [],
  "transaction": Array [],
  "transactionInputOutput": Array [],
  "transfer": Array [],
}
`;
